### PR TITLE
CM-82 EventEngine integration

### PR DESCRIPTION
**Pull request by VladSemenyuk**

### CODE REVIEW

**Comment by VladSemenyuk** [comment]
@bonewell

File: src/components/ivdcm_module/include/ivdcm_module/event_engine/event.h [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+   * @brief Retrieves event message
+   *
+   * @return The message received in event
+   */
+  inline const EventMessage& event_message() const;
+
+  /*
+   * @brief Retrieves event message request ID
+   */
+  virtual int32_t event_message_function_id() const = 0;
```
Comment (DKlimenko) [inline src/components/ivdcm_module/include/ivdcm_module/event_engine/event.h]: not sure if it must be virtual

File: src/components/ivdcm_module/include/ivdcm_module/request_controller.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_IVDCM_MODULE_INCLUDE_IVDCM_MODULE_REQUEST_CONTROLLER_H_
+#define SRC_COMPONENTS_IVDCM_MODULE_INCLUDE_IVDCM_MODULE_REQUEST_CONTROLLER_H_
+
+#include <map>
+
+#include "can_cooperation/commands/command.h"
+#include "can_cooperation/can_module_timer.h"
```
Comment (DKlimenko) [inline src/components/ivdcm_module/include/ivdcm_module/request_controller.h]: can_cooperation?

**Comment by DKlimenko** [comment]
reviewed

File: src/components/ivdcm_module/include/ivdcm_module/request_controller.h (hunk 2 above)
Comment (VladSemenyuk) [inline src/components/ivdcm_module/include/ivdcm_module/request_controller.h]: Review wasn't modified, I'll modify it when add ivdcm command and timer

File: src/components/ivdcm_module/CMakeLists.txt [hunk 3]
```diff
@@ -34,6 +34,8 @@ include_directories (
 
 set (SOURCES
     src/ivdcm.cc
+    #src/request_controller.cc
+    src/ivdcm_event.cc
```
Comment (bonewell) [inline src/components/ivdcm_module/CMakeLists.txt]: I think name event.cc will be better.

File: src/components/ivdcm_module/include/ivdcm_module/event_engine/event.h [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+/*template<typename EventMessage, typename EventID>
+void Event<EventMessage, EventID>::raise() {
+  EventDispatcher<EventMessage, EventID>::instance()->raise_event(*this);
+}*/
+
+}  // namespace event_engine
+
+#endif  // SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_EVENT_ENGINE_EVENT_H_
```
Comment (bonewell) [inline src/components/ivdcm_module/include/ivdcm_module/event_engine/event.h]: Update comment, please, no CAN_COOPERATION

File: src/components/ivdcm_module/include/ivdcm_module/event_engine/event.h [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+#define SRC_COMPONENTS_IVDCM_MODULE_INCLUDE_IVDCM_MODULE_EVENT_ENGINE_EVENT_H_
+
+#include <string>
+
+namespace event_engine {
+
+template<typename EventMessage, typename EventID>
+class Event {
+ public:
+  /*
```
Comment (bonewell) [inline src/components/ivdcm_module/include/ivdcm_module/event_engine/event.h]: For Doxygen blocks of comment you need to start block using double stars /**

File: src/components/ivdcm_module/include/ivdcm_module/event_engine/event.h [hunk 6]
```diff
@@ -0,0 +1,120 @@
+/*
+ Copyright (c) 2013, Ford Motor Company
```
Comment (bonewell) [inline src/components/ivdcm_module/include/ivdcm_module/event_engine/event.h]: Update copyright please.

File: src/components/ivdcm_module/include/ivdcm_module/event_engine/event_dispatcher.h [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+#define  SRC_COMPONENTS_IVDCM_MODULE_INCLUDE_IVDCM_MODULE_EVENT_ENGINE_EVENT_DISPATCHER_H_
+
+#include <list>
+#include <map>
+
+#include "utils/lock.h"
+#include "utils/singleton.h"
+
+#include "can_cooperation/event_engine/event.h"
+#include "can_cooperation/event_engine/event_observer.h"
```
Comment (bonewell) [inline src/components/ivdcm_module/include/ivdcm_module/event_engine/event_dispatcher.h]: Don't forget to change these lines :)

File: src/components/ivdcm_module/src/request_controller.cc [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+#include "ivdcm_module/request_controller.h"
+#include "json/json.h"
+#include "utils/logger.h"
+#include "functional_module/timer/timer_director_impl.h"
+#include "functional_module/settings.h"
+
+namespace ivdcm_module {
+namespace request_controller {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "CANRequestController")
```
Comment (bonewell) [inline src/components/ivdcm_module/src/request_controller.cc]: Don't forget to fix it :)

**Comment by bonewell** [comment]
Reviewed.

**Pull request merged**